android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -300,6 +361,84 @@ private MediaSource buildMediaSource(Uri uri, String overrideExtension) {
+
+        if (drmParam == null) {
+            return null;
+        }
+
+        DefaultDrmSessionManager<FrameworkMediaCrypto> drmSessionManager = null;
+        final String drmLicenseUrl = drmParam.getLicensingServerUrl();
+        final HashMap<String, String> keyRequestProperties = Utils.getParams(drmParam);
+
+        UnsupportedDrmException exception = null;
+        if (Util.SDK_INT < 18) {
+            throw new UnsupportedDrmException(0);
+        } else {
+            UUID drmSchemeUuid = Util.getDrmUuid(drmParam.getDrmScheme());
+            if (drmSchemeUuid == null) {
+                throw new UnsupportedDrmException(UnsupportedDrmException.REASON_UNSUPPORTED_SCHEME);
+            } else {
+                drmSessionManager = buildDrmSessionManagerV18(drmSchemeUuid, drmLicenseUrl, keyRequestProperties, false);
+
+                final String offline = drmParam.getOfflineLicense();
+
+                if (offline != null) {
+                    drmSessionManager.setMode(DefaultDrmSessionManager.MODE_QUERY, Base64.decode(offline, Base64.NO_WRAP));
+                }
+            }
+        }
+        return drmSessionManager;
+    }
+
+    private void handleDrmError(UnsupportedDrmException exception) {
+        final int errorStringId;
+        switch (exception.reason) {
+            case 0:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_not_supported;
+                break;
+            case UnsupportedDrmException.REASON_UNSUPPORTED_SCHEME:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_unsupported_scheme;
+                break;
+            default:
+                errorStringId = com.dice.shield.downloads.R.string.error_drm_unknown;
+                break;
+        }
+
+        final String errorString = getContext().getString(errorStringId);
+        eventEmitter.error(errorString, exception);
+    }
+
+    private DefaultDrmSessionManager<FrameworkMediaCrypto> buildDrmSessionManagerV18(
+            UUID uuid, String licenseUrl, HashMap<String, String> requestProperties, boolean multiSession)
+            throws UnsupportedDrmException {
+        HttpDataSource.Factory licenseDataSourceFactory =
+                (DlmWrapper.getInstance(getContext().getApplicationContext())).buildHttpDataSourceFactory(/* listener= */ null);
+        HttpMediaDrmCallback drmCallback =
+                new HttpMediaDrmCallback(licenseUrl, licenseDataSourceFactory);
+        if (requestProperties != null) {
+            for (String key : requestProperties.keySet()) {
+                drmCallback.setKeyRequestProperty(key, requestProperties.get(key));
+            }
+        }
+        releaseMediaDrm();
+        mediaDrm = FrameworkMediaDrm.newInstance(uuid);
+        this.drmCallback = drmCallback;
+        return new DefaultDrmSessionManager<>(uuid, mediaDrm, drmCallback, null, multiSession);
+    }
+
+    private void releaseMediaDrm() {
```

Comment:
Do not think we have multi-threaded operations running here, but it feels like that release should be a synchronize method.

Reply:
It should be done on one thread only so I don't think it is needed. However I can see it is not called when player is released and I guess it should be.